src/viking.js

```diff
@@ -1,11 +1,89 @@
 // Soldier
-class Soldier {}
-
+class Soldier {
+    constructor(health,strength){
+this.health=health;
+this.strength=strength;
+    }
+    attack(){
+        return this.strength
+    }
+    receiveDamage(damage){
+        this.health-=damage
+    }  
+}
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name,health,strength){
+    super(name,health,strength)
+    this.name=name
+    this.health=health
+    this.strength=strength
+}
+receiveDamage(damage){
+    this.health-=damage
+    if(this.health>0){
+        return(`${this.name} has received ${damage} points of damage`)
+    }
+    else {
+return(`${this.name} has died in act of combat`)
+    }
+}
+battleCry(){
+    return "Odin Owns You All!"
+}
+}
+
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+    receiveDamage(damage){
+        this.health-=damage
+        if(this.health>0){
+            return(`A Saxon has received ${damage} points of damage`)
+        }
+        else {
+    return(`A Saxon has died in combat`)
+        }
+    }
+}
 
 // War
-class War {}
+class War {
+    vikingArmy=[]
+    saxonArmy=[]
+    addViking(Viking){
+        this.vikingArmy.push(Viking)
+    }
+addSaxon(Saxon){
+     this.saxonArmy.push(Saxon)
+}
+vikingAttack(){
+   let saxonRandom=Math.floor(Math.random()*this.saxonArmy.length)
+   let vikingRandom=Math.floor(Math.random()*this.vikingArmy.length)
+let fight=this.saxonArmy[saxonRandom].receiveDamage(this.vikingArmy[vikingRandom].strength)
+if (this.saxonArmy[saxonRandom].health<=0) {
+    this.saxonArmy.splice([saxonRandom],1)
+}
+return fight
+}
+saxonAttack(){
+    let saxonRandom=Math.floor(Math.random()*this.saxonArmy.length)
+   let vikingRandom=Math.floor(Math.random()*this.vikingArmy.length)
+    let fight2=this.vikingArmy[vikingRandom].receiveDamage(this.saxonArmy[saxonRandom].strength)
+    if (this.vikingArmy[saxonRandom].health<=0) {
+        this.vikingArmy.splice([vikingRandom],1)
+}
+return fight2
+}
+showStatus(){
+    if (this.saxonArmy<=0){
+        return "Vikings have won the war of the century!"
+    }
+    else if (this.vikingArmy<=0) {
+return "Saxons have fought for their lives and survived another day..."
+    }
+    else if(this.saxonArmy=this.vikingArmy){
```

Comment:
Here you want to compare the vikingArray with the saxonArray but you're doing it with just one `=` this is used to reassign a new value to a variable, if you want to compare something you need to use `==` or `===`
```
(...)

    else if(this.saxonArmy === this.vikingArmy){
        return "Vikings and Saxons are still in the thick of battle."
    }

(...)
```
But this is not what you are asked to do in the lab instructions. They want you to print this message if there are still vikings and saxons alive, so the correct condition is:
```
(...)

else if (this.saxonArmy.length >= 1 && this.vikingArmy.length >= 1) {
      return "Vikings and Saxons are still in the thick of battle.";
    }

(...)
```